src/sql/workbench/contrib/searchViewPane/browser/searchWidget/searchWidgetService.ts

```diff
@@ -0,0 +1,297 @@
+	onSearchSubmit(options: any): void;
+	onSearchCancel({ focus: boolean }): void;
+}
+export interface ISearchWidgetService {
+	searchWidgets: Map<string, any>;
+	registerSearchWidget(container: HTMLElement, pane: ViewPaneContainer, params: ISearchViewPaneOptions): void;
+	getSearchWidget(VIEWLET_ID: string): any | undefined;
+	getSearchResultsView(parentConatiner: ViewPaneContainer): any | undefined;
+}
+export class SearchWidgetService extends Disposable implements ISearchWidgetService {
+
+	declare readonly _serviceBrand: undefined;
+	searchWidgets: Map<string, any>;
+	constructor(
+		@IInstantiationService private readonly _instantiationService: IInstantiationService,
+		@IViewDescriptorService protected viewDescriptorService: IViewDescriptorService,
+	) {
+		super();
+		this.searchWidgets = new Map<string, any>();
+	}
+
+	getSearchWidget(VIEWLET_ID: string): any | undefined {
+		return this.searchWidgets.get(VIEWLET_ID);
+	}
+
+	registerSearchWidget(container: HTMLElement, pane: ViewPaneContainer, params: ISearchViewPaneOptions): void {
+		if (!this.searchWidgets.get(params.VIEWLET_ID)) {
+			let searchWidget = this._register(this._instantiationService.createInstance(SimpleSearchWidget, container, <IViewExplorerSearchOptions>{
+				value: '',
+				replaceValue: undefined,
+				isRegex: false,
+				isCaseSensitive: false,
+				isWholeWords: false,
+				searchHistory: [],
+				replaceHistory: [],
+				preserveCase: false,
+				showSearchResultsPane: params.showSearchResultsPane,
+			}, pane, params.VIEWLET_ID));
+
+			this.searchWidgets.set(params.VIEWLET_ID, searchWidget);
+			if (params.showSearchResultsPane) {
+				this.registerSearchResultsView(params);
+
+				//register search results view specific stuff.
+				if (searchWidget.searchView) {
+					this._register(searchWidget.onDidHeightChange(() => searchWidget.searchView?.reLayout()));
+
+					this._register(searchWidget.onPreserveCaseChange(async (state) => {
+						if (searchWidget.searchView?.searchViewModel) {
+							searchWidget.searchView.searchViewModel.preserveCase = state;
+							await searchWidget.searchView.refreshTree();
+						}
+					}));
+
+					this._register(searchWidget.searchInput.onInput(() => searchWidget.searchView?.updateActions()));
+				}
+			}
+
+			if (params.onSearchSubmit) {
+				searchWidget.onSearchSubmit(options => params.onSearchSubmit(options));
+
+				searchWidget.searchInput.onDidOptionChange((options) => params.onSearchSubmit(options));
+			}
+
+			if (params.onSearchCancel) {
+				searchWidget.onSearchCancel((focus) => params.onSearchCancel(focus));
+			}
+		}
+	}
+
+	registerSearchResultsView(params: ISearchViewPaneOptions) {
+		let viewDescriptors = [];
+		switch (params.VIEWLET_ID) {
+			case 'workbench.view.notebooks': viewDescriptors.push({
+				id: NotebookSearchResultsView.ID,
+				name: localize('notebookExplorer.searchResults', "Search Results"),
+				ctorDescriptor: new SyncDescriptor(NotebookSearchResultsView),
```

Comment:
This should be a generic SearchResultsView

Reply:
this is specific to notebooks viewlet, I'm not creating a generic results view. (at least not for this iteration)